src/DynamoCoreWpf/UI/GuidedTour/CustomRichTextBox.cs

```diff
@@ -69,113 +72,124 @@ private static FlowDocument GetCustomDocument(string Text)
+                    }
 
-                if (bHyperlinkActive)
-                {
-                    //Finding the end character that indicates the hyperlink is complete (no empty spaces are allowed in hyperlink URL just in the name)
-                    if (word.Contains("="))
+                    //A format for inserting a hyperlink between text was found              
+                    if (word.StartsWith("#"))
+                    {
+                        bHyperlinkActive = true;
+                    }
+
+                    if (bHyperlinkActive)
+                    {
+                        //Finding the end character that indicates the hyperlink is complete (no empty spaces are allowed in hyperlink URL just in the name)
+                        if (word.Contains("="))
+                        {
+                            string linkURL = word.Split('=')[1];
+                            hyperlinkName += word.Split('=')[0];
+
+                            Run run3 = new Run(hyperlinkName.Replace("#", ""));
+                            Hyperlink link = new Hyperlink(run3);
+                            link.IsEnabled = true;
+                            link.NavigateUri = new Uri(linkURL);
+                            link.RequestNavigate += (sender, args) => Process.Start(args.Uri.ToString());
+                            para.Inlines.Add(link);
+                            bHyperlinkActive = false;
+                        }
+                        //The hyperlink name is the next word followed by the # char (empty spaces are allowed) and the URL value is the one followed after the = char
+                        else
+                            hyperlinkName += word.Replace("#", "") + " ";
+                    }
+                    else if (bBoldActive)
                     {
-                        string linkURL = word.Split('=')[1];
-                        hyperlinkName += word.Split('=')[0];
-
-                        Run run3 = new Run(hyperlinkName.Replace("#", ""));
-                        Hyperlink link = new Hyperlink(run3);
-                        link.IsEnabled = true;
-                        link.NavigateUri = new Uri(linkURL);
-                        link.RequestNavigate += (sender, args) => Process.Start(args.Uri.ToString());
-                        para.Inlines.Add(link);
-                        bHyperlinkActive = false;
+                        string boldText = word;
+                        boldText = boldText.Replace("**", "");
+                        para.Inlines.Add(new Bold(new Run(boldText)));
+                    }
+                    else if (bImageActive)
+                    {
+                        //this will contatenate each word so at the end we will have the full image path
+                        imageName += word;
+                    }
+                    else if (bBulletListActive == true)
+                    {
+                        bulletEntryText += (word + " ");
```

Comment:
can you put comments as to why adding a space here?

Reply:
@QilongTang  comments added in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12250/commits/c3995980180bb91f906ebbe6eb891924d0a584ea